### DISCUSSION TITLE
Plugins in the improved plugin system

### DISCUSSION THREAD

**Discussion by Gedeen** [WinMerge/winmerge#948] [Q&A] [answered]
The help file tells us that a plugin is a single dll or a scriptlet (sct) file, but now there is also a file called Plugins.xml in the MergePlugin folder. This is not covered in the help file (at least I couldn't find it). It seems to me that it is now possible to use an external command as an unpacker plugin. Even a command that executes a vb script which is included in Plugins.xml (although is is commented out).

Is there any discription about this Plugins.xml file, what the use cases can be and how to use it ?

I think this feature is quite handy i.e. to use a sed script before comparing two files.

Greetings

**Comment by sdottaka** [comment, marked as answer]
Plugins.xml is currently undocumented.

It is not recommended that users write to C:\Program Files\WinMerge\MergePlugins\Plugins.xml, as the next update installation will overwrite the Plugins.xml file.

If you create the file %APPDATA%\WinMerge\MergePlugins\Plugins.xml, WinMerge will read it and will not overwrite it in the next update installation.

**Reply by Gedeen** [reply]
Ok, thanks. This means that if I use my own Plugins.xml in appdata I will not receive any updates of this files in next releases and I have to take over any changes of this file into my own Plugins.xml. Of course we have an excellent compare and merge tool at hand ;-)
